cloud/openstack/os_project_facts.py

```diff
@@ -0,0 +1,164 @@
+     required: false
+'''
+
+EXAMPLES = '''
+# Gather facts about previously created projects
+- os_projects_facts:
+    cloud: awesomecloud
+- debug: var=openstack_projects
+
+# Gather facts about a previously created project by name
+- os_projects_facts:
+    cloud: awesomecloud
+    name: demoproject
+- debug: var=openstack_projects
+
+# Gather facts about a previously created project in a specific domain
+- os_projects_facts
+    cloud: awesomecloud
+    name: demoproject
+    domain: admindomain
+- debug: var=openstack_projects
+
+# Gather facts about a previously created project in a specific domain
+  with filter
+- os_projects_facts
+    cloud: awesomecloud
+    name: demoproject
+    domain: admindomain
+    filters:
+      enabled: False
+- debug: var=openstack_projects
+'''
+
+
+RETURN = '''
+openstack_projects:
+    description: has all the OpenStack facts about projects
+    returned: always, but can be null
+    type: complex
+    contains:
+        id: 
+            description: Unique UUID.
+            returned: success
+            type: string
+        name:
+            description: Name given to the project.
+            returned: success
+            type: string
+        description:
+            description: Description of the project
+            returned: success
+            type: string
+        enabled:
+            description: Flag to indicate if the project is enabled
+            return: success
+            type: bool
+        domain_id:
+            description: Domain ID containing the project
+            return: success
+            type: bool
+'''
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=False, default=None),
+        domain=dict(required=False, default=None),
+        filters=dict(required=False, type='dict', default=None),
+    )
+
+    module = AnsibleModule(argument_spec)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    try:
+        name = module.params['name']
+        domain = module.params['domain']
+        filters = module.params['filters']
+
+        if domain:
+            opcloud = shade.operator_cloud(**module.params)
+            try:
+                # We assume admin is passing domain id
+                dom = opcloud.get_domain(domain)['id']
+                domain = dom
+            except:
+                # If we fail, maybe admin is passing a domain name.
+                # Note that domains have unique names, just like id.
+                try:
+                    dom = opcloud.search_domains(filters={'name': domain})[0]['id']
+                    domain = dom
+                except:
+                    # Ok, let's hope the user is non-admin and passing a sane id
+                    pass
+
+            if not filters:
+                filters = {}
+
+            filters['domain_id'] = domain
+
+        cloud = shade.openstack_cloud(**module.params)
```

Comment:
Hmm, yeah, OperatorCloud subclasses OpenStackCloud.
Which makes me think I could later refactor os_project (I did some copy-pasta from it here).